sdk/core/azure-core/test/nlohmann-json-test/src/unit-algorithms.cpp

```diff
@@ -0,0 +1,337 @@
+/*
```

Comment:
I am going to assume everything under `test/nlohmann-json-test/src/` is a **verbatim** copy of the files from https://github.com/nlohmann/json/tree/v3.8.0/test/src from the `3.8.0` git tag, and skip reviewing this. Please let me know if that's not the case or if you made any changes to it.

Reply:
The only changes are:
- replacing `#include<nlohmann/json.hpp>` for `#include<azure/core/internal/json.hpp>`
- replacing `using nlohmann::json json` for `using Azure::Core::Internal::Json json`

And some other tests were removed because it's removed/unsupported features that nlohmaan offer with global macros like `_json` .  Instead of having our own version `_az_json`, I just remove this feature. It is kind of nice to have IMO